APCAppCore/APCAppCore/Library/Categories/NSDate+Helper.h

```diff
@@ -77,6 +94,138 @@ extern NSString * const NSDateDefaultDateFormat;
+ calculation will use 30 days per month (every month) and
+ 365 days per year.  If you specify a real date, the
+ calculations will account for the actual number of days
+ per month during the elapsed months, and will account for
+ leap years.
+ 
+ @see +timeIntervalBySubtractingISO8601Duration:fromDate
+ @see http://en.wikipedia.org/wiki/ISO_8601#Durations
+ */
++ (NSTimeInterval) timeIntervalByAddingISO8601Duration: (NSString *) duration
+                                                toDate: (NSDate *) date;
+
+/**
+ Just like +timeIntervalByAddingISO8601Duration:toDate:,
+ but subtracts the specified duration from the specified
+ date.
+ 
+ @see +timeIntervalByAddingISO8601Duration:toDate:
+ */
++ (NSTimeInterval) timeIntervalBySubtractingISO8601Duration: (NSString *) duration
+                                                   fromDate: (NSDate *) date;
+
+/**
+ Just like +timeIntervalByAddingISO8601Duration:toDate:,
+ operating on self instead of a passed-in date parameter.
+
+ @see +timeIntervalByAddingISO8601Duration:toDate:
+ */
+- (NSTimeInterval) timeIntervalByAddingISO8601Duration: (NSString *) duration;
+
+/**
+ Just like +timeIntervalByAddingISO8601Duration:toDate:,
+ but subtracts the specified duration from the specified
+ self instead of adding to a passed-in parameter.
+
+ @see +timeIntervalByAddingISO8601Duration:toDate:
+ */
+- (NSTimeInterval) timeIntervalBySubtractingISO8601Duration: (NSString *) duration;
+
+/**
+ Adds the specified duration string to the specified
+ date.  Duration strings follow the ISO 8601 standard.
+ Some examples:
+
+ @code
+ P3D        3 days ("P" == "period")
+ P1W        1 week
+ P2D5H10S   2 days, 5 hours, and 10 seconds
+ P36H       36 hours, i.e., a day and a half
+ P1.5D      a day and a half, i.e., 36 hours
+ @endcode
+
+ If you want to include minutes, you'll need a (T)ime
+ indicator somewhere to the left of it, to distinguish it
+ from months.  Like this:
+
+ @code
+ P3DT5M     3 days + 5 minutes
+ @endcode
+
+ You can use any time interval from (S)econds through
+ (Y)ears, except fortnights.  Time will be measured in
+ terms of the interval:  1.5D and 36H both give you a
+ day and a half, but 1M or 1Y will give you different
+ numbers of days, depending on the months and leap years
+ as measured from startDate.
+ 
+ This method calls
+ +timeIntervalByAddingISO8601Duration:toDate.
+
+ @param durationString An ISO 8601 duration string, as
+ described above.
+
+ @see -dateBySubtractingISO8601Duration:
+ @see +timeIntervalByAddingISO8601Duration:toDate:
+ @see http://en.wikipedia.org/wiki/ISO_8601#Durations
+ */
+- (NSDate *) dateByAddingISO8601Duration: (NSString *) durationString;
+
+/**
+ Just like -dateByAddingISO8601Duration:, but subtracts
+ the specified duration from self.  See
+ -dateByAddingISO8601Duration: for details.
+
+ @see -dateByAddingISO8601Duration:
+ */
+- (NSDate *) dateBySubtractingISO8601Duration: (NSString *) durationString;
 
```

Comment:
Created a [GitHub issue](https://github.com/ResearchKit/AppCore/issues/75) for this.